src/osd/ReplicatedPG.cc

```diff
@@ -10462,6 +10423,11 @@ void ReplicatedPG::hit_set_trim(RepGather *repop, unsigned max)
   for (unsigned num = updated_hit_set_hist.history.size(); num > max; --num) {
     list<pg_hit_set_info_t>::iterator p = updated_hit_set_hist.history.begin();
     assert(p != updated_hit_set_hist.history.end());
+    if (!p->using_gmt) {
```

Comment:
This would mean we can't delete hit set objects created before the upgrade?  That doesn't sound great.